### PR TITLE
Core24 357 frontend display validation response from api

### PR DESCRIPTION
**Pull request by neb42**
Jira ticket: JIRA-CORE24-357

### CODE REVIEW

File: libs/prm-components/src/lib/components/EntityDetailForm.component.tsx [hunk 1]
```diff
@@ -82,6 +84,18 @@ export const EntityDetailForm: React.FC<Props> = ({
     };
   }, [form.formState.isDirty, readOnly]);
 
+  useEffect(() => {
+    if (error) {
+      error.issues.forEach((issue) => {
+        console.log(issue);
```
Comment (dfigueroa87) [inline libs/prm-components/src/lib/components/EntityDetailForm.component.tsx:90]: Maybe we should add the `no-console` eslint rule?

File: libs/prm-components/src/lib/pages/EntityDetailPage/EntityDetailPage.component.tsx [hunk 2]
```diff
@@ -54,12 +56,32 @@ export const EntityDetailPage: React.FC<Props> = ({ mode }) => {
...
+    }
+    return '';
+  })();
+
+  const validationError = (() => {
+    if (error && isAxiosError(error) && error.response?.status === 400) {
+      return new ZodError(error.response.data.errors);
+    }
+    return null;
+  })();
```
Comment (dfigueroa87) [inline libs/prm-components/src/lib/pages/EntityDetailPage/EntityDetailPage.component.tsx:77]: Why using self invoked functions?

Maybe I'm just not used to them, but I find this way easier to read:
```
const validationError = error && isAxiosError(error) && error.response?.status === 400 ? new ZodError(error.response.data.errors) : null;
```

File: libs/prm-components/src/lib/pages/EntityDetailPage/EntityDetailPage.component.tsx [hunk 3]
```diff
@@ -54,12 +56,32 @@ export const EntityDetailPage: React.FC<Props> = ({ mode }) => {
     }
   }, [mode, entityType]);
 
+  const errorMessage = (() => {
+    if (error) {
+      if (isAxiosError(error)) {
+        if (error.response?.status === 400) {
+          return 'There was an error with the data you submitted. Please check the form and try again.';
+        }
+        return `Internal error, contact support. [${error.response?.status}] [${error.response?.data?.message}]`;
```
Comment (internetti) [inline libs/prm-components/src/lib/pages/EntityDetailPage/EntityDetailPage.component.tsx:65]: if you tell them to contact support, you should at least add an email address, or the link to the helpform

File: libs/prm-components/src/lib/pages/EntityDetailPage/EntityDetailPage.component.tsx (hunk 2 above)
Comment (neb42) [reply libs/prm-components/src/lib/pages/EntityDetailPage/EntityDetailPage.component.tsx:77]: In this case you're probably right.

Generally I like them when setting a variable with conditional logic, but a separate function feels like overkill.

File: libs/prm-components/src/lib/pages/EntityDetailPage/EntityDetailPage.component.tsx [hunk 4]
```diff
@@ -54,12 +57,46 @@ export const EntityDetailPage: React.FC<Props> = ({ mode }) => {
...
+              There was an error with the data you submitted. Please check the
+              form and try again.';
+            </Text>
+          );
+        }
+        return (
+          <Text>
+            Internal error,{' '}
+            <Link
+              to="https://nrc.freshservice.com/a/catalog/request-items/95"
```
Comment (dfigueroa87) [inline libs/prm-components/src/lib/pages/EntityDetailPage/EntityDetailPage.component.tsx:75]: This is the link for core v1, no?

I guess not a big deal at the moment, but we should eventually get it changed

**Review by dfigueroa87 — APPROVED** [review]

File: libs/prm-components/src/lib/pages/EntityDetailPage/EntityDetailPage.component.tsx (hunk 4 above)
Comment (internetti) [reply libs/prm-components/src/lib/pages/EntityDetailPage/EntityDetailPage.component.tsx:80]: more importantly, this is the one that didn't work, it should be this one:
https://nrc.freshservice.com/support/catalog/items/95

**Review by internetti — CHANGES_REQUESTED** [review]

**Review by internetti — CHANGES_REQUESTED** [review]
when I correct all the errors in the form, the save button doesn't go back to active mode

**Comment by neb42** [comment]
> when I correct all the errors in the form, the save button doesn't go back to active mode

Do you have steps to reproduce this? I can't

**Review by internetti — APPROVED** [review]

**Pull request merged**